### ISSUE TITLE
CouchDB 3.1.0 not starting because it uses wrong DLLs on Windows 10

### ISSUE THREAD

**Issue by magel20** [apache/couchdb#3309]
## Description

We are using CouchDB as local database within a project. Basically it is just a small program which accesses the database. That's all working fine on my machine, but on an second machine we are having problems with starting up CouchDB at all. Installation works fine but then the CouchDB does not start on the machine.

The problem seems to be very similar to the one described here: https://github.com/apache/couchdb/issues/2680

Exactly the same DLLs are causing the same problems. We are already using CouchDB version 3.1.0 so we already use the fix mentioned in the above issue.

As far as I have analyzed the problem seems to be that the mentioned DLLs are also stored in the C:/Windows/System32 folder and CouchDB uses them instead of the files in the CouchDB/bin folder. If I remove the DLLs from System32 folder CouchDB starts as expected.

The Fix shipped out with CouchDB 3.0.1 sets the Systems32 to the path so I think this is causing the problem?

## Steps to Reproduce

I cannot reproduce it on my machine. The problem only appears on the second machine.

## Expected Behaviour

CouchDB starting on both machines

## Your Environment

* CouchDB version used: 3.1.0
* Browser name and version: Irrelevant, CouchDB does not start at all
* Operating system and version: Windows 10 Version 1909 (Build 18363.1198)

**Comment by wohali** [comment]
HI there,

https://github.com/apache/couchdb/pull/2710 was merged into 3.x a while ago. It should place `%COUCHDB_BIN_DIR%` ahead of `%SystemRoot%\system32` in the path.

Can you confirm that your `couchdb.cmd` file has this change in place?

**Comment by magel20** [comment]
Hey,

yes I can confirm that the 'couchdb.cmd' has the changes from https://github.com/apache/couchdb/pull/2710

I also tried it with CouchDB 3.1.1 as well but the problem still occurs.

**Comment by wohali** [comment]
@magel20 Unfortunately, I can't confirm this in our development environment.

That `PATH` environment variable change explicitly puts the CouchDB binary directory ahead of the `C:\Windows\System32` directory. The Windows dynamic linker searches PATH in order to find the necessary DLLs it needs. It can't run with only `%COUCHDB_BIN_DIR%, it needs the system's DLLs as well to run.

What happens if you open an Administrator Command Prompt and run the `couchdb.cmd` rather than using the automatic Windows Service? Does the problem still occur?

Do you know which program placed those DLLs in your `%SYSTEMROOT%` directory? Arguably, *that's* the bug here, because with those DLLs in there, if the `%PATH%` tweak doesn't prevent those from being accessed, there's little we can do here to fix it. Can you move those DLLs to the directory of the program that actually needs them?

**Comment by magel20** [comment]
I tried the `couchdb.cmd` in an Administrator Command Prompt as well as the Windows Service. Problem occurred both times.

So the `set PATH=...` command inside the `couchdb.cmd` sets  the CouchDB binary directory ahead of the others. Should I be able to see that change in the `Environment Variables` UI in Windows 10? Because there I do not see the `%COUCHDB_BIN_DIR%` in the PATH at all or is this set done by the script only temporary?

If I understand it correctly the problem I encounter should not occur because the `couchdb.cmd` first looks inside the CouchDB binary directory finds the DLLs and should then not care about the DLLs in the `%SYSTEMROOT%` directory? Might there be a possibility that the set command is not working? Or is it more likely that the `%PATH%` tweak does not prevent the DLLs from being accessed if they are placed in the `%SYSTEMROOT%` ?

Unfortunately I have not found out which program placed the DLLs there. Removing them from `%SYSTEMROOT%` made the CouchDB work and so far no other program I use on a daily basis has complained about it. But if I got you correct it should not be the case that a program places its DLLs in the `%SYSTEMROOT%` directory and that is the actual bug?
